tests/components/hooks.test.ts

```diff
@@ -641,6 +641,56 @@ describe("hooks", () => {
       ]);
     });
 
+    test("effect types are inferred from dependencies", async () => {
+      // @ts-ignore (declared but never used)
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      class MyComponent extends Component {
```

Comment:
`@ts-ignore` required for jest
`// eslint-disable-next-line` required for linting
Don't know if there is another way